src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -94,7 +93,7 @@ public override void Inject<T>(PropagationContext context, T carrier, Action<T,
                     continue;
                 }
 
-                baggage.Append(WebUtility.UrlEncode(item.Key)).Append('=').Append(WebUtility.UrlEncode(item.Value)).Append(',');
+                baggage.Append(Uri.EscapeDataString(item.Key)).Append('=').Append(Uri.EscapeDataString(item.Value)).Append(',');
```

Comment:
According to the spec only the value should be encoded. More:
- https://github.com/w3c/baggage/issues/131
- https://github.com/open-telemetry/opentelemetry-go/issues/4946

However, I think it would be better to handle it as a separate issue and PR.

Reply:
I created https://github.com/open-telemetry/opentelemetry-dotnet/issues/5479

Reply:
@pellared - Looks like there are multiple issues we need to address here. The test harness is not implemented in .NET, I think we SHOULD first get that in before fixing these issues on ad hoc basis. Ideally, We SHOULD fix them all together in one release so that the users can see all changes together. 

As per our offline discussion, You did mention there was a discussion to not used the test harness as a reference. Once you find that, Could you please add a link to issue/discussion? Thanks!. Also, it would be great if you and @lachmatt  could join one if the SIG meeting to discuss this in detail.

Reply:
> I think we SHOULD first get that in before fixing these issues on ad hoc basis. Ideally, We SHOULD fix them all together in one release so that the users can see all changes together.

Agree.

> Also, it would be great if you and @lachmatt could join one if the SIG meeting to discuss this in detail.

We will do our best to join today.

> As per our offline discussion, You did mention there was a discussion to not used the test harness as a reference. Once you find that, Could you please add a link to issue/discussion?

I was wrong. We still decided to use the test harness. This is the PR which I was recalling: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4937. And it was about Trace Context - not Baggage.